### PR TITLE
test_init conversion

### PR DESCRIPTION
**Pull request by MichelleArk**
resolves: #6010

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: my_project_test16742493171360845266_test_init/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Welcome to your new dbt project!
```
Comment (stu-k) [inline my_project_test16742493171360845266_test_init/README.md:1]: Did you mean to include this project at the root level of the repo?

File: tests/functional/init/test_init.py [hunk 2]
```diff
@@ -0,0 +1,688 @@
...
+  target: my_target
+"""
+            )
+
+
+class TestInitInvalidProfileTemplate:
+    @mock.patch("dbt.task.init._get_adapter_plugin_names")
+    @mock.patch("click.confirm")
+    @mock.patch("click.prompt")
+    def test_init_task_in_project_with_invalid_profile_template(
```
Comment (ChenyuLInx) [inline tests/functional/init/test_init.py:312]: So this test would get the `profiles.yml` that exisits in `tests/functional/init`?

File: core/dbt/task/init.py [hunk 3]
```diff
@@ -175,6 +175,7 @@ def create_profile_from_target(self, adapter: str, profile_name: str):
         profile_template_path = adapter_path / "profile_template.yml"
 
         if profile_template_path.exists():
+            # import pdb; pdb.set_trace()
```
[the remark this answers is not in the document]
Comment (MichelleArk) [reply core/dbt/task/init.py:178]: oopsie

File: my_project_test16742493171360845266_test_init/README.md (hunk 1 above)
Comment (MichelleArk) [reply my_project_test16742493171360845266_test_init/README.md:1]: no, not at all. my bad 😵‍💫

**Review by ChenyuLInx — APPROVED** [review]
LGTM!

File: tests/functional/init/test_init.py (hunk 2 above)
Comment (MichelleArk) [reply tests/functional/init/test_init.py:312]: This test would have a profiles.yml created for the `project` fixture in `{project.project_root}/{project.profiles_dir}/profiles.yml` to start, which gets overwritten by invoking `dbt init` during the test.

**Pull request merged**